### PR TITLE
refactor: split Zoe types.js into separate files for Zoe Service, Contract Facet

### LINKED ISSUE #1498: [Zoe] Split type files into zoeService and ContractFacet?
For the source files, the following split has worked great:
   `src/zoeService/` vs `src/contractFacet/` vs directly in `src/` for what they share

For the type files, I appreciate that we don't want to break them up into corresponding smaller files. But it would be clarifying to break each up into these three sections clearly labeled. Without some kind of marker like that (or a naming convention) I find I'm always first asking "where does this go?"

_Originally posted by @erights in https://github.com/_render_node/MDE3OlB1bGxSZXF1ZXN0UmV2aWV3NDY0NTMyNDQw/pull_request_reviews/more_threads_

### PR DESCRIPTION
**Pull request by katelynsills**
Split types.js into contractFacet/types, zoeService/types, and contractSupport/types. Shared definitions stay.

Sometimes it was difficult to decide which file the typedef should belong in, so I'm open to rearrangement.

Closes #1498

### CODE REVIEW

**Review by erights — APPROVED** [review]
LGTM. Thanks!

File: packages/zoe/src/internal-types.js [hunk 1]
```diff
@@ -211,3 +211,32 @@
...
+ * @typedef RootAndAdminNode
+ * @property {Object} root
+ * @property {AdminNode} adminNode
+ */
+
+/**
+ * @typedef {Object} AdminNode
+ * A powerful object that can be used to terminate the vat in which a contract
+ * is running, to get statistics, or to be notified when it terminates. The
+ * VatAdmin object is only available to the contract from within the contract so
```
Comment (michaelfig) [inline packages/zoe/src/internal-types.js]:
```suggestion
 * object is only available from within the contract so
```

**Review by michaelfig — APPROVED** [review]
LGTM!

File: packages/zoe/src/contractFacet/types.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+ * @property {MakeInvitation} makeInvitation
+ * @property {() => void} shutdown
+ * @property {() => ZoeService} getZoeService
+ * @property {() => Issuer} getInvitationIssuer
+ * @property {() => Terms } getTerms
+ * @property {(issuer: Issuer) => Brand} getBrandForIssuer
+ * @property {(brand: Brand) => Issuer} getIssuerForBrand
+ * @property {GetAmountMath} getAmountMath
+ * @property {MakeZCFMint} makeZCFMint
+ * @property {(exit: ExitRule= }) => ZcfSeatKit} makeEmptySeatKit
```
Comment (Chris-Hibbert) [inline packages/zoe/src/contractFacet/types.js]: This feels really picky, but how much does lint care about trailing white space within the braces? I see that `Terms` and `ExitRule=` are followed by a blank before the close brace.  It's a little visually distracting.

**Review by Chris-Hibbert — APPROVED** [review]
This seems like a fine division. Thanks.

I had one small picky comment. It was probably in the original code.  Feel free to ignore.

File: packages/zoe/src/contractFacet/types.js (hunk 2 above)
Comment (erights) [reply packages/zoe/src/contractFacet/types.js]:
```js
 * @property {(exit: ExitRule= }) => ZcfSeatKit} makeEmptySeatKit
```
mismatched brackets?

**Pull request merged**